packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -395,19 +438,25 @@ class _BottomNavigationTile extends StatelessWidget {
     // =======
     if (showSelectedLabels && !showUnselectedLabels) {
       bottomPadding = Tween<double>(
-        begin: 0.0,
-        end: selectedFontSize / 2.0,
+        begin: selectedIconDiff / 2.0,
+        end: selectedFontSize / 2.0 - unselectedIconDiff / 2.0,
       ).evaluate(animation);
       topPadding = Tween<double>(
-        begin: selectedFontSize,
-        end: selectedFontSize / 2.0,
+        begin: selectedFontSize + selectedIconDiff / 2.0,
+        end: selectedFontSize / 2.0 - unselectedIconDiff / 2.0,
       ).evaluate(animation);
     }
 
     // Center all icons if no labels are shown.
     if (!showSelectedLabels && !showUnselectedLabels) {
-      bottomPadding = 0.0;
-      topPadding = selectedFontSize;
+      bottomPadding = Tween<double>(
+        begin: selectedIconDiff / 2.0,
+        end: unselectedIconDiff / 2.0,
+      ).evaluate(animation);
+      topPadding = Tween<double>(
+        begin: selectedFontSize + selectedIconDiff / 2.0,
+        end: selectedFontSize + unselectedIconDiff / 2.0,
```

Comment:
Same comment here, is this unselected font size?

Reply:
See above.